### PR TITLE
Add the ability to specify which boosters to not operate on

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #78

### CODE REVIEW

File: for-all-boosters.sh [hunk 1]
```diff
@@ -841,7 +842,7 @@ show_help () {
     simple_log "    -f                            Bypass check for local changes, forcing execution if changes exist."
     simple_log "    -h                            Display this help message."
     simple_log "    -l                            Specify where the local copies of the boosters should be found. Defaults to current working directory."
-    simple_log "    -m                            The boosters to operate on (comma separated value). The name of each booster can either be the full booster name, or the simple booster name (for example: circuit-breaker) Not selecting this option means that all boosters will be operated on."
+    simple_log "    -m                            The boosters to operate on (comma separated value). Boosters are effectively white-listed in this mode. The name of each booster can is the simple booster name (for example: circuit-breaker). Using '-' in front of the names inverts the booster selection (boosters are effectively black-listed)"
```
Comment (gytis) [inline for-all-boosters.sh]: Typo "can is".

File: for-all-boosters.sh [hunk 2]
```diff
@@ -1112,13 +1140,18 @@ do
...
+        # 2) the user explicitly included the booster using it's simple name (the part without 'spring-boot-' and '-booster')
+        # 3) the user did not include the simple booster name in the explicitly excluded boosters
+        should_process=true
+        if (( ${selection_type} > 0 )) && ! element_in "${booster_simple_name}" "${explicitly_selected_boosters[@]}"; then
+          should_process=false
+        elif (( ${selection_type} < 0 )) && element_in "-${booster_simple_name}" "${explicitly_selected_boosters[@]}"; then
+          should_process=false
+        fi
+
+        if [ "$should_process" = true ] ; then
```
Comment (gytis) [inline for-all-boosters.sh]: `if [[ ${should_process} ]]; then`?

Comment (geoand) [reply for-all-boosters.sh]: That would probably work, I'll check it out

Comment (geoand) [reply for-all-boosters.sh]: It seems to work, but is not recommended: https://stackoverflow.com/a/21210966/2504224

Comment (gytis) [reply for-all-boosters.sh]: My bad, I thought it's the same as in Java.

**Comment by metacosm** [comment]
What do you think of using only one `-` to exclude, instead of needing to repeat it for all the boosters we want to exclude? Initially, when I didn't think too much about this, I thought we might want to select some and exclude some but that's stupid… :) 
So maybe `-m -cache,http` would exclude both `cache` and `http` boosters? I think that would simplify the code and make it easier on the user, no?

**Comment by geoand** [comment]
I kind of think that current way of needing to repeat the sign for all is the best way to prevent the user for getting the wrong behavior.

I do understand the rationale behind your proposal but I think execution safety should be the top concern.
@gytis WDYT?

**Comment by metacosm** [comment]
In that case, maybe we shouldn't conflate both behavior in a single option and use `-x` to exclude boosters? Though, then we'd need to make these options exclusive.

**Comment by geoand** [comment]
@metacosm That is another option we could follow. My preference list is (higher to lower):

* As is done now
* Using the `-x` option
* Only needing to specify the `-` on the first booster

**Comment by metacosm** [comment]
My preference would be `-x` first, then only needing the first `-` then as it is now :) 
I really think that having a separate option will make for cleaner code overall and less cognitive load for users.

**Comment by geoand** [comment]
@metacosm We need a tie breaker :P

**Comment by metacosm** [comment]
@gytis to the rescue!

**Comment by gytis** [comment]
I think that a separate option like `-x` would be the clearest. I cannot vouch for the implementation simplicity though.

**Comment by geoand** [comment]
I'll update the implementation to use `-x`.
It shouldn't complicate things too much

**Comment by geoand** [comment]
Implementation changed to use `-x`

**Comment by geoand** [comment]
Could you please review again @gytis @metacosm to account for the new `-x` implementation?

**Pull request merged**